js/modules/k6/http/request_test.go

```diff
@@ -1212,6 +1214,57 @@ func TestSystemTags(t *testing.T) {
 	}
 }
 
+func TestRequestCompression(t *testing.T) {
+	t.Parallel()
+	tb, state, _, rt, _ := newRuntime(t)
+	defer tb.Cleanup()
+
+	// We don't expect any failed requests
+	state.Options.Throw = null.BoolFrom(true)
+
+	var text = `
+	Lorem ipsum dolor sit amet, consectetur adipiscing elit.
+	Maecenas sed pharetra sapien. Nunc laoreet molestie ante ac gravida.
+	Etiam interdum dui viverra posuere egestas. Pellentesque at dolor tristique,
+	mattis turpis eget, commodo purus. Nunc orci aliquam.`
+
+	tb.Mux.HandleFunc("/compressed-text", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		require.Equal(t, r.Header.Get("Content-Encoding"), "gzip")
+
+		expectedLength, err := strconv.Atoi(r.Header.Get("Content-Length"))
+		require.NoError(t, err)
+
+		var compressedBuf bytes.Buffer
+		n, err := io.Copy(&compressedBuf, r.Body)
+		require.Equal(t, int(n), expectedLength)
+		require.NoError(t, err)
+
+		g, err := gzip.NewReader(&compressedBuf)
+		require.NoError(t, err)
+
+		var buf bytes.Buffer
+		_, err = io.Copy(&buf, g)
+		require.NoError(t, err)
+		require.Equal(t, text, buf.String())
+	}))
+
+	t.Run("gzip", func(t *testing.T) {
+		_, err := common.RunString(rt, tb.Replacer.Replace(`
+		http.post("HTTPBIN_URL/compressed-text", `+"`"+text+"`"+`,  {"compression": "gzip"});
+	`))
+		require.NoError(t, err)
+	})
+
+	t.Run("unsupported compression", func(t *testing.T) {
+		_, err := common.RunString(rt, tb.Replacer.Replace(`
+		http.post("HTTPBIN_URL/compressed-text", `+"`"+text+"`"+`,  {"compression": "George"});
+	`))
+		require.Error(t, err)
+		require.Equal(t, err.Error(),
+			`GoError: unknown compression algorithm George, only supported algorithm is gzip`)
```

Comment:
Can we remove this check, or make it be `string starts with "GoError: unknown compression algorithm"`? As soon as we add a new compression algorithm we'll also have to needlessly fix this test, since the current error message is so overly specific.

Reply:
The overly specificness is so that we can tell them what they can use. I can
1. remove it from the error -have it only in the docs
2. remove the check - any error will make the test to pass
3. make `require.Equal` but [require.Contains](https://godoc.org/github.com/stretchr/testify/require#Contains) - I prefer this option the most

Reply:
I'm fine with having the current specific error message be as it is.

I just dislike super-fragile tests that we know we're going go break, so I only want the test to be fixed. Option 3, whether using `require.Contains()` or `require.True(strings.HasPrefix(...))` will be plenty good enough.